diff --git a/src/v4l2_camera.cpp b/src/v4l2_camera.cpp
index 64b4399..8ba5071 100644
--- a/src/v4l2_camera.cpp
+++ b/src/v4l2_camera.cpp
@@ -56,7 +56,9 @@ V4L2Camera::V4L2Camera(rclcpp::NodeOptions const & options)
     return;
   }
 
-  cinfo_ = std::make_shared<camera_info_manager::CameraInfoManager>(this, camera_->getCameraName());
+  cinfo_ = std::make_shared<camera_info_manager::CameraInfoManager>(
+    get_node_base_interface(), get_node_services_interface(),
+    get_node_logging_interface(), camera_->getCameraName());
 
   parameters_.declareDeviceParameters(*camera_);
 
