diff --git a/src/imu_display.cpp b/src/imu_display.cpp
index d693612..8b76de2 100644
--- a/src/imu_display.cpp
+++ b/src/imu_display.cpp
@@ -113,7 +113,7 @@ void ImuDisplay::reset()
     acc_visual_->hide();
 }
 
-void ImuDisplay::update(float /* dt */, float /* ros_dt */)
+void ImuDisplay::update(std::chrono::nanoseconds /* dt */, std::chrono::nanoseconds /* ros_dt */)
 {
     updateTop();
     updateBox();
diff --git a/src/imu_display.h b/src/imu_display.h
index 7dc2598..d8fc3ce 100644
--- a/src/imu_display.h
+++ b/src/imu_display.h
@@ -69,7 +69,7 @@ class ImuDisplay
 
     void reset() override;
 
-    void update(float dt, float ros_dt) override;
+    void update(std::chrono::nanoseconds dt, std::chrono::nanoseconds ros_dt) override;
 
   private:
     void createProperties();
diff --git a/src/mag_display.cpp b/src/mag_display.cpp
index 6fb21cc..07495f3 100644
--- a/src/mag_display.cpp
+++ b/src/mag_display.cpp
@@ -85,7 +85,7 @@ void MagDisplay::reset()
     mag_visual_->hide();
 }
 
-void MagDisplay::update(float /* dt */, float /* ros_dt */)
+void MagDisplay::update(std::chrono::nanoseconds /* dt */, std::chrono::nanoseconds /* ros_dt */)
 {
     updateMag();
 }
diff --git a/src/mag_display.h b/src/mag_display.h
index 8b262b2..5c0980b 100644
--- a/src/mag_display.h
+++ b/src/mag_display.h
@@ -67,7 +67,7 @@ class MagDisplay
 
     void reset() override;
 
-    void update(float dt, float ros_dt) override;
+    void update(std::chrono::nanoseconds dt, std::chrono::nanoseconds ros_dt) override;
 
   private:
     void createProperties();
