diff --git a/zstd_point_cloud_transport/CMakeLists.txt b/zstd_point_cloud_transport/CMakeLists.txt
index 51e2d6c..278c591 100644
--- a/zstd_point_cloud_transport/CMakeLists.txt
+++ b/zstd_point_cloud_transport/CMakeLists.txt
@@ -9,6 +9,8 @@ find_package(pluginlib REQUIRED)
 find_package(point_cloud_interfaces REQUIRED)
 find_package(point_cloud_transport REQUIRED)
 find_package(rclcpp REQUIRED)
+find_package(zstd_cmake_module REQUIRED)
+find_package(zstd REQUIRED)
 
 set(dependencies
   pluginlib
@@ -29,7 +31,7 @@ target_link_libraries(${PROJECT_NAME} PRIVATE
   ${point_cloud_interfaces_TARGETS}
   point_cloud_transport::point_cloud_transport
   rclcpp::rclcpp
-  zstd
+  zstd::zstd
 )
 
 target_include_directories(${PROJECT_NAME} PRIVATE
diff --git a/zstd_point_cloud_transport/include/zstd_point_cloud_transport/zstd_subscriber.hpp b/zstd_point_cloud_transport/include/zstd_point_cloud_transport/zstd_subscriber.hpp
index b5828d3..cfea8f0 100644
--- a/zstd_point_cloud_transport/include/zstd_point_cloud_transport/zstd_subscriber.hpp
+++ b/zstd_point_cloud_transport/include/zstd_point_cloud_transport/zstd_subscriber.hpp
@@ -51,8 +51,6 @@ class ZstdSubscriber
 public:
   ZstdSubscriber();
 
-  std::string getTransportName() const override;
-
   void declareParameters() override;
 
   std::string getDataType() const override;
