diff --git a/zstd_point_cloud_transport/CMakeLists.txt b/zstd_point_cloud_transport/CMakeLists.txt
index 51e2d6c..332290e 100644
--- a/zstd_point_cloud_transport/CMakeLists.txt
+++ b/zstd_point_cloud_transport/CMakeLists.txt
@@ -9,12 +9,16 @@ find_package(pluginlib REQUIRED)
 find_package(point_cloud_interfaces REQUIRED)
 find_package(point_cloud_transport REQUIRED)
 find_package(rclcpp REQUIRED)
+find_package(zstd_cmake_module REQUIRED)
+find_package(zstd REQUIRED)
 
 set(dependencies
   pluginlib
   point_cloud_interfaces
   point_cloud_transport
   rclcpp
+  zstd_cmake_module
+  zstd
 )
 
 add_library(${PROJECT_NAME}
@@ -29,7 +33,7 @@ target_link_libraries(${PROJECT_NAME} PRIVATE
   ${point_cloud_interfaces_TARGETS}
   point_cloud_transport::point_cloud_transport
   rclcpp::rclcpp
-  zstd
+  zstd::zstd
 )
 
 target_include_directories(${PROJECT_NAME} PRIVATE
diff --git a/zstd_point_cloud_transport/include/zstd_point_cloud_transport/zstd_subscriber.hpp b/zstd_point_cloud_transport/include/zstd_point_cloud_transport/zstd_subscriber.hpp
index b5828d3..cfea8f0 100644
--- a/zstd_point_cloud_transport/include/zstd_point_cloud_transport/zstd_subscriber.hpp
+++ b/zstd_point_cloud_transport/include/zstd_point_cloud_transport/zstd_subscriber.hpp
@@ -51,8 +51,6 @@ class ZstdSubscriber
 public:
   ZstdSubscriber();
 
-  std::string getTransportName() const override;
-
   void declareParameters() override;
 
   std::string getDataType() const override;
diff --git a/zstd_point_cloud_transport/package.xml b/zstd_point_cloud_transport/package.xml
index 17fc04a..fa0e39b 100644
--- a/zstd_point_cloud_transport/package.xml
+++ b/zstd_point_cloud_transport/package.xml
@@ -22,7 +22,7 @@
   <depend>point_cloud_interfaces</depend>
   <depend>point_cloud_transport</depend>
   <depend>rclcpp</depend>
-  <depend>libzstd-dev</depend>
+  <depend>zstd_cmake_module</depend>
 
   <test_depend>ament_lint_auto</test_depend>
   <test_depend>ament_lint_common</test_depend>
